Source/Weavers/Unity.Cecil.Visitor/TypeReferenceExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Mono.Cecil;
+
+namespace Unity.Cecil.Visitor
+{
+    public static class TypeReferenceExtensions
+    {
+        public static int InheritanceChainLength(TypeReference type)
```

Comment:
a little more dotnetty way to do this would be an extension method `IEnumerable<TypeReference> SelectInheritanceChain(TypeReference @this)` and then the caller can call Count() on that.